apis/applicationintegration/v1alpha1/sqs_types.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// Enum values for Queue attribute names
+const (
+	AttributeAll                                   string = "All"
+	AttributePolicy                                string = "Policy"
+	AttributeVisibilityTimeout                     string = "VisibilityTimeout"
+	AttributeMaximumMessageSize                    string = "MaximumMessageSize"
+	AttributeMessageRetentionPeriod                string = "MessageRetentionPeriod"
+	AttributeApproximateNumberOfMessages           string = "ApproximateNumberOfMessages"
+	AttributeApproximateNumberOfMessagesNotVisible string = "ApproximateNumberOfMessagesNotVisible"
+	AttributeCreatedTimestamp                      string = "CreatedTimestamp"
+	AttributeLastModifiedTimestamp                 string = "LastModifiedTimestamp"
+	AttributeQueueArn                              string = "QueueArn"
+	AttributeApproximateNumberOfMessagesDelayed    string = "ApproximateNumberOfMessagesDelayed"
+	AttributeDelaySeconds                          string = "DelaySeconds"
+	AttributeReceiveMessageWaitTimeSeconds         string = "ReceiveMessageWaitTimeSeconds"
+	AttributeRedrivePolicy                         string = "RedrivePolicy"
+	AttributeDeadLetterQueueARN                    string = "DeadLetterQueueARN"
+	AttributeMaxReceiveCount                       string = "MaxReceiveCount"
+	AttributeFifoQueue                             string = "FifoQueue"
+	AttributeContentBasedDeduplication             string = "ContentBasedDeduplication"
+	AttributeKmsMasterKeyID                        string = "KmsMasterKeyId"
+	AttributeKmsDataKeyReusePeriodSeconds          string = "KmsDataKeyReusePeriodSeconds"
+)
+
+// Tag is a key value pairs attached to a Amazon SQS queue.
+type Tag struct {
+
+	// The key name that can be used to look up or retrieve the associated value.
+	Key string `json:"key"`
+
+	// The value associated with a key in a tag.
+	Value string `json:"value,omitempty"`
```

Comment:
This has an `omitempty` tag, but isn't a pointer or marked `// +optional`. Is it an optional field?

Reply:
I have updated this this code section, was some reason it is not being reflected here.

Reply:
@sahil-lakhwani I can see the new `// +optional` tag but it still seems that this is a `string` and not a `*string`. Are you saying you've made that change but it's not showing up in GitHub for some reason?

Reply:
Okay I had missed changing that field to pointer type. I have changed it in the new commit. Thanks for pointing that!